### PR TITLE
feat: list event valid submission and record to database

### PR DESCRIPTION
**Pull request by hunjixin**
### Review Type Requested (choose one):
- [ ] **Glance** - superficial check (from domain experts)
- [ ] **Logic** - thorough check (from everybody doing review)

add a subcommand to record pow submission to pg database

resolve https://github.com/Lilypad-Tech/internal/issues/100

### Summary
Provide a one line summary and link to any relevant references

### Task/Issue reference

Closes: add_link_here

### Details (optional)
Add any additional details that might help Code Reviewers digest this PR

### CODE REVIEW

**Comment by hunjixin** [comment]
./lilypad  collection-pow-submission -n dev --web3-private-key 0x8b3a350cf5c34c9194ca85829a2df0ec3153be0318b5e2d3348e872092edffba --pg-connection-string <pg dsn>

table structure

```
                                             Table "public.pow_valid_pow_submitteds"
       Column       |       Type        | Collation | Nullable | Default | Storage  | Compression | Stats target | Description
--------------------+-------------------+-----------+----------+---------+----------+-------------+--------------+-------------
 id                 | character varying |           | not null |         | extended |             |              |
 challenge          | character varying |           |          |         | extended |             |              |
 wallet_address     | character varying |           |          |         | extended |             |              |
 node_id            | character varying |           |          |         | extended |             |              |
 nonce              | character varying |           |          |         | extended |             |              |
 start_timestamp    | bigint            |           |          |         | plain    |             |              |
 complete_timestamp | bigint            |           |          |         | plain    |             |              |
 difficulty         | character varying |           |          |         | extended |             |              |
Indexes:
    "pow_valid_pow_submitteds_pkey" PRIMARY KEY, btree (id)
    "address_index" btree (wallet_address)
```

```
                                 id                                |                            challenge                             |               wallet_address               |                    node_id                     |                                     nonce                                      | start_timestamp | complete_timestamp |                               difficulty
------------------------------------------------------------------+------------------------------------------------------------------+--------------------------------------------+------------------------------------------------+--------------------------------------------------------------------------------+-----------------+--------------------+------------------------------------------------------------------------
 e02a9e812204b3581cb695795927ca1cbc37c3b265f84406f00839d149124613 | 3257cba5340860b98fdf146f6b4ed71d0a545c15edbcb2b19d641ef72f91f835 | 0x15d34AAf54267DB7D7c367839AAf71A00a2C6A65 | Qmbkz9U7Nq81vcgzTdb3mYaxLuJVZLDcSRPrLtoJFhByCQ | 627569                                                                         |      1718683337 |         1718683345 | 2021876946624580403476403221115901569728113256934678191056046655411322
 d6ba2d9c3975faeedd4f828816c91c3bbacffe42da596f976ab623064f08e2af | b9214c4d9f60efdb3764e034ace504b7493f613ec7c0e13b2a7da707a6283b12 | 0x15d34AAf54267DB7D7c367839AAf71A00a2C6A65 | Qmbkz9U7Nq81vcgzTdb3mYaxLuJVZLDcSRPrLtoJFhByCQ | 101318078082651670995624611882601919371611236582435493534525386006923995388032 |      1718683642 |         1718683735 | 2021876946624580403476403221115901569728113256934678191056046655411322
 e5d750784de924d4d5a07dd895fe2554bffa1f6374090d5f2e62fdc219dbffdd | 7306737e7a0c19131ce66c1ab8f882c4a7c311937035a3755a017e4ff368603f | 0x15d34AAf54267DB7D7c367839AAf71A00a2C6A65 | Qmbkz9U7Nq81vcgzTdb3mYaxLuJVZLDcSRPrLtoJFhByCQ | 43422033463993573283839119378257965444976244249615211514796594002967424265005  |      1718683783 |         1718683792 | 2021876946624580403476403221115901569728113256934678191056046655411322
 e53b313d72519535df494dc4350db885c80a6a56b04e2d06ca8af1abc2c38de6 | 5bfc20b94e20a28d08a58c621c18b3888e76719905c86444a302065225a5dde4 | 0x15d34AAf54267DB7D7c367839AAf71A00a2C6A65 | Qmbkz9U7Nq81vcgzTdb3mYaxLuJVZLDcSRPrLtoJFhByCQ | 43422033463993573283839119378257965444976244249615211514796594002967426616818  |      1718683815 |         1718683855 | 2021876946624580403476403221115901569728113256934678191056046655411322
 d0acb1dbb29764b62dc35494560eb9fbf8a20a94a9396cd72f6056428c08c707 | 78dbf6c219cd2ae2f11ee89965f022b72de10e4199686b3620ffa44f884499a1 | 0x15d34AAf54267DB7D7c367839AAf71A00a2C6A65 | Qmbkz9U7Nq81vcgzTdb3mYaxLuJVZLDcSRPrLtoJFhByCQ | 101318078082651670995624611882601919371611236582435493534525386006923988677905 |      1718683871 |         1718683874 | 2021876946624580403476403221115901569728113256934678191056046655411322
```

File: pkg/collectpowsubmission/collect_submission.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+		}
+
+		if len(ids) < pageSize {
+			break
+		}
+
+		offset += pageSize
+	}
+
+	// Get submissions from sc and check them with bloomfilter and database. if mssing, record it
```
Comment (bgins) [inline pkg/collectpowsubmission/collect_submission.go]:
```suggestion
	// Get submissions from sc and check them with bloomfilter and database. if missing, record it
```

File: cmd/lilypad/collection-pow-submission.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import (
+	"github.com/lilypad-tech/lilypad/pkg/collectpowsubmission"
+	"github.com/lilypad-tech/lilypad/pkg/options"
+	optionsfactory "github.com/lilypad-tech/lilypad/pkg/options"
+	"github.com/lilypad-tech/lilypad/pkg/system"
+	"github.com/lilypad-tech/lilypad/pkg/web3"
+	"github.com/spf13/cobra"
+)
+
+func collectionPowSubmissionCmd() *cobra.Command {
```
Comment (bgins) [inline cmd/lilypad/collection-pow-submission.go]: Could we follow the existing pattern for naming commands and call this `newCollectionPowSubmissionCmd`? Event better, we would like to start using the convention for initialisms/acronyms mentioned here: https://go.dev/wiki/CodeReviewComments#initialisms, which would change the name to `newCollectionPOWSubmissionCmd`.

**Review by bgins — COMMENTED** [review]
Left a few initial comments. 

Also, could we rename this PR to `feat: list event valid submission and record to database`. The PR will be squashed and merged with the PR title as the commit message, and we plan to use conventional commits for semantic versioning: https://www.conventionalcommits.org/en/v1.0.0/

File: cmd/lilypad/collection-pow-submission.go (hunk 2 above)
Comment (hunjixin) [reply cmd/lilypad/collection-pow-submission.go]: done

**Comment by hunjixin** [comment]
merge before next network update

**Review by richardbremner — CHANGES_REQUESTED** [review]
@hunjixin I appreciate this work, thank you!

There's an issue I see though, I don't think @AquiGorka would approve this PR, because it's writing direct to the Postgres db from lilypad, and his architecture is to log the events, and use the cron-server to read those logs and transform them into db writes. 

Basically what was documented here for this process https://www.notion.so/lilypadnetwork/PoW-Data-Capture-c7797381694b4403b61b4198377a3abb?pvs=4

**Comment by hunjixin** [comment]
this pr not work for old pow construct, i may need close this one

**Comment by taoshengshi** [comment]
> this pr not work for old pow construct, i may need close this one

@AquiGorka 

1. If yes for do not upgrade the SC, let us close this PR as the reason above

2. Is using AWS EFS mandatory? refer https://www.notion.so/lilypadnetwork/PoW-Data-Capture-c7797381694b4403b61b4198377a3abb

**Comment by AquiGorka** [comment]
> Is using AWS EFS mandatory? refer https://www.notion.so/lilypadnetwork/PoW-Data-Capture-c7797381694b4403b61b4198377a3abb

Yes, this is the architecture we are following.

**Pull request closed without merging**